aml/src/statement.rs

```diff
@@ -135,11 +143,11 @@ where
                         .map(move |then_branch| Ok((predicate_arg.as_bool()?, then_branch)))
                 })
                 .then(choice!(
```

Comment:
This is fine as the fix to this, but I wonder if a cleaner solution would be to replace the whole `choice!(..., id().map(&[])` construct with a wrapper combinator (e.g. `maybe`) that would catch either a `WrongParser` or an `UnexpectedEndOfStream`? I think that would work